src/Functions/URL/cutToFirstSignificantSubdomain.cpp

```diff
@@ -43,34 +43,155 @@ using FunctionCutToFirstSignificantSubdomainWithWWWRFC = FunctionStringToString<
+    FunctionDocumentation::Syntax cutToFirstSignificantSubdomainWithWWW_syntax = "cutToFirstSignificantSubdomainWithWWW(url)";
+    FunctionDocumentation::Arguments cutToFirstSignificantSubdomainWithWWW_arguments = {
+        {"url", "URL or domain string to process.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue cutToFirstSignificantSubdomainWithWWW_returned_value = {
+        "Returns the part of the domain from the first significant subdomain up to the top-level domain, preserving 'www.'", 
+        {"String"}
+    };
+    FunctionDocumentation::Examples cutToFirstSignificantSubdomainWithWWW_examples = {
+    {
+        "Preserves www prefix when present", 
+        "SELECT cutToFirstSignificantSubdomainWithWWW('https://www.news.bbc.co.uk/')", 
+        "www.bbc.co.uk"
+    },
+    {
+        "Returns domain without www when not present", 
+        "SELECT cutToFirstSignificantSubdomainWithWWW('https://docs.github.com/')", 
+        "github.com"
+    },
+    {
+        "Handles simple www domains", 
+        "SELECT cutToFirstSignificantSubdomainWithWWW('www.example.org')", 
+        "www.example.org"
+    }
+    };
+    FunctionDocumentation::IntroducedIn cutToFirstSignificantSubdomainWithWWW_introduced_in = {20, 12};
+    FunctionDocumentation::Category cutToFirstSignificantSubdomainWithWWW_category = FunctionDocumentation::Category::URL;
+    FunctionDocumentation cutToFirstSignificantSubdomainWithWWW_documentation = {cutToFirstSignificantSubdomainWithWWW_description, cutToFirstSignificantSubdomainWithWWW_syntax, cutToFirstSignificantSubdomainWithWWW_arguments, cutToFirstSignificantSubdomainWithWWW_returned_value, cutToFirstSignificantSubdomainWithWWW_examples, cutToFirstSignificantSubdomainWithWWW_introduced_in, cutToFirstSignificantSubdomainWithWWW_category};
+    
+    factory.registerFunction<FunctionCutToFirstSignificantSubdomainWithWWW>(cutToFirstSignificantSubdomainWithWWW_documentation);
+
+    FunctionDocumentation::Description cutToFirstSignificantSubdomainRFC_description = R"(
+Similar to [`cutToFirstSignificantSubdomain`](#cutToFirstSignificantSubdomain) but follows stricter rules to be compatible with [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986).
+
+This variant performs more thorough URL parsing according to RFC 3986 standards, which may result in lower performance 
+compared to the standard version but provides more accurate handling of edge cases.
+    )";
+    FunctionDocumentation::Syntax cutToFirstSignificantSubdomainRFC_syntax = "cutToFirstSignificantSubdomainRFC(url)";
+    FunctionDocumentation::Arguments cutToFirstSignificantSubdomainRFC_arguments = {
+        {"url", "URL or domain string to process according to RFC 3986.", {"String"}}
+    };
+    FunctionDocumentation::ReturnedValue cutToFirstSignificantSubdomainRFC_returned_value = {
+        "Returns the part of the domain from the first significant subdomain up to the top-level domain, without 'www.', following RFC 3986.", 
```

Comment:
```suggestion
        "Returns the part of the domain that includes top-level subdomains up to the first significant subdomain if possible, otherwise returns an empty string.", 
```